public/app/plugins/panel/barchart/module.tsx

```diff
@@ -145,6 +145,11 @@ export const plugin = new PanelPlugin<BarChartOptions, BarChartFieldConfig>(BarC
           max: 1,
           step: 0.01,
         },
+      })
+      .addFieldNamePicker({
+        path: 'colorByField',
+        name: 'Color by field',
+        description: 'Use the color value for a sibling field to color each bar value.',
       });
```

Comment:
Yes -- I think we should still explore the general sibling color approach in https://github.com/grafana/grafana/pull/41381, however having a simple panel option will cover the most common use cases.

Also the panel options approach means we can transparently setup a fieldConfig flavor when that works (later!)

Reply:
i'm also excited about the general approach. it allows the color to represent a third dimension of bars rather than being purely stylistic or just re-enforcing the y value position (in terms of threshold colors).

this also may transfer nicely to mapping colors per-point for x/y or maybe heatmap block colors based on bucket counts.

Reply:
but still think we should do it via color option we already have:
https://github.com/grafana/grafana/pull/41520

Reply:
The approach we have is pretty esoteric -- especially for this case where it really only makes sense as an override.  I think we *could* make it work, but that requires passing index everywhere and keeping closures with the sibling frames.  I'm nervous that is a big ask in general.

Reply:
yea, the passing index everywhere and the display processor function needs to capture the field from the outer scope (so in streaming it will use old values), and the problem with the caching display processor (we can solve that via caching on index+value but still a bit messy). So maybe this is enough of an edge case so we only really need it in bar chart and a making it a platform feature is not needed

Reply:
> So maybe this is enough of an edge case so we only really need it in bar chart and a making it a platform feature is not needed

For now, i think this approach is fine.